src/main/java/com/minecraftabnormals/abnormals_core/core/mixin/EndBiomeProviderMixin.java

```diff
@@ -17,63 +16,43 @@
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
 
-import java.util.List;
-
 @Mixin(EndBiomeProvider.class)
-public abstract class EndBiomeProviderMixin extends BiomeProvider {
-	@Shadow
-	@Final
-	private SimplexNoiseGenerator generator;
+public abstract class EndBiomeProviderMixin {
 	@Shadow
 	@Final
 	private Registry<Biome> lookupRegistry;
 	@Shadow
 	@Final
-	private Biome theEndBiome;
-	@Shadow
-	@Final
-	private Biome endHighlandsBiome;
-	@Shadow
-	@Final
 	private Biome endMidlandsBiome;
 	@Shadow
 	@Final
-	private Biome smallEndIslandsBiome;
+	private Biome endBarrensBiome;
 	@Shadow
 	@Final
-	private Biome endBarrensBiome;
-
+	private Biome endHighlandsBiome;
+	
 	private Layer noiseBiomeLayer;
-
-	private EndBiomeProviderMixin(List<Biome> biomes) {
-		super(biomes);
-	}
-
-	@Inject(at = @At("RETURN"), method = "<init>")
-	private void init(Registry<Biome> lookupRegistry, long seed, CallbackInfo info) {
+	
+	@Inject(at = @At("RETURN"), method = "<init>(Lnet/minecraft/util/registry/Registry;JLnet/minecraft/world/biome/Biome;Lnet/minecraft/world/biome/Biome;Lnet/minecraft/world/biome/Biome;Lnet/minecraft/world/biome/Biome;Lnet/minecraft/world/biome/Biome;)V")
+	private void init(Registry<Biome> lookupRegistry, long seed, Biome theEndBiome, Biome endHighlandsBiome, Biome endMidlandsBiome, Biome smallEndIslandsBiome, Biome endBarrensBiome, CallbackInfo ci) {
 		this.noiseBiomeLayer = ACLayerUtil.createEndBiomeLayer(lookupRegistry, (seedModifier) -> new LazyAreaLayerContext(25, seed, seedModifier));
 	}
-
-	@Inject(at = @At("HEAD"), method = "getNoiseBiome(III)Lnet/minecraft/world/biome/Biome;", cancellable = true)
+	
+	@Inject(at = @At("RETURN"), method = "getNoiseBiome(III)Lnet/minecraft/world/biome/Biome;", cancellable = true)
 	private void addEndBiomes(int x, int y, int z, CallbackInfoReturnable<Biome> info) {
-		int i = x >> 2;
-		int j = z >> 2;
-		if ((long) i * (long) i + (long) j * (long) j <= 4096L) {
-			info.setReturnValue(this.theEndBiome);
-		} else {
-			float noise = EndBiomeProvider.getRandomNoise(this.generator, i * 2 + 1, j * 2 + 1);
+		if (
+				info.getReturnValue().equals(endHighlandsBiome) ||
+						info.getReturnValue().equals(endMidlandsBiome) ||
+						info.getReturnValue().equals(endBarrensBiome)
+		) {
```

Comment:
`info.getReturnValue()` should be moved to a local variable.
This if statement is far too simple to make up five lines.
I don't see a reason to use `equals()` here because there will only ever be one of these biomes.
```suggestion
		Biome returnValue = info.getReturnValue();
		if (returnValue == this.endHighlandsBiome || returnValue == this.endMidlandsBiome || returnValue == this.endBarrensBiome) {
```
The formatting of this suggestion might be a little messed up.